### ISSUE TITLE
Failed to create temporary file for long destination filename

### ISSUE THREAD

**Issue by dabrain34** [ccache/ccache#593]
### How to reproduce ###

1. Building gst-build with Meson buildsystem
2. 
3. 

### Actual behavior ###

An error occurs with this message
```
ccache: error: Failed to create temporary file for subprojects/gst-plugins-bad/tests/check/4f868b3@@libs_nalutils@exe/_home_scerveau_DEV_tmp_gst-build_subprojects_gst-plugins-bad_tests_check_.._.._gst-libs_gst_codecparsers_nalutils.c.o: File name too long
```

### Expected behavior ###

Not expecting Filename too long as an error for 117 filename len.

### Environment ###

ccache version 3.4.1

Ubuntu 18.04 x86_64
gcc 7.5.0
ext4 fs

**Comment by jrosdahl** [comment]
This happens because ccache writes to the destination file via a temporary file and the temporary filename gets is longer than the destination filename. It is already fixed on master but I'll backport it as a bug fix to the 3.7.x series.

**Comment by dabrain34** [comment]
txh for the feedback. Good to know ....
